enhancements/console/samples.md

```diff
@@ -0,0 +1,309 @@
+
+Currently, this opens a form to import a git repository. Other sample types aren't supported at the moment.
+
+## Motivation
+
+Creating a `ConsoleSample` CRD will promote the samples feature in the Developer Console to an extendable feature similar to QuickStarts. This makes it much easier for other teams to provide their own samples, for example with their operator.
+
+Samples are no longer tied to the exact OpenShift version and could be updated with the operator. Also customers can add their own samples if needed.
+
+This will also allow us to migrate from the (deprecated) [samples-operator](https://github.com/openshift/cluster-samples-operator) to a simple custom resource in the future and to add other sample types in the future.
+
+### User Stories
+
+* As an OpenShift Serverless engineer, I want to provide additional samples that are tied to an operator version, not an OpenShift release.
+* As a cluster administrator, I want to add additional samples to the Developer Console sample catalog.
+
+### Goals
+
+1. This feature should allow other teams to provide their Developer Console samples without applying them to the sample-operator or the console-operator.
+2. Samples should be allowed to localize/translated.
+
+### Non-Goals
+
+1. We don't want to turn off the support for the (deprecated) [samples-operator](https://github.com/openshift/cluster-samples-operator) right now. So the console will not provide any samples on its own (in 4.14).
+2. A way to hide installed samples. Operators creating a sample resource should have a way to opt out of their samples.
+
+## Proposal
+
+### Workflow Description
+
+The Console Operator will provide a cluster-wide `ConsoleSample` CRD.
+
+Other operators like the OpenShift Serverless operator will install their own `ConsoleSample` CRs.
+
+The web console will automatically load all `ConsoleSample` CRs and show them in the sample catalog.
+
+Users can select one of samples and follow the import flows to create a demo application.
+
+### API Extensions
+
+The console operator automatically installs a new `ConsoleSample` CRD.
+
+### Risks and Mitigations
+
+#### The OpenShift Console is optional and could be disabled
+
+Other operators should expect that the `ConsoleSample` CRD is unavailable like the other `Console*` CRDs.
+
+### Drawbacks
+
+Another extra (but specific) custom resource just for console samples.
+
+## Design Details
+
+An "Import from Git" example of how we have it today as part of the Builder Images annotation:
+
+```yaml
+apiVersion: console.openshift.io/v1
+kind: ConsoleSample
+metadata:
+  name: java-sample
+spec:
+  title: Java
+  description: |
+    Build and run Java applications using Maven and OpenJDK.
+    Sample repository: https://github.com/jboss-openshift/openshift-quickstarts
+  icon: base64 encoded image
+  provider: Red Hat
+  badge: Serverless function
+  tags:
+  - java
+  - jboss
+  - openjdk
+  source:
+    type: GitImport
```

Comment:
Would be good to discuss what source types are available and how they're used.

Reply:
Added a new section:

> ### Source types
> 
> The `source` attribute is an union type and will support only two `type`s in 4.14:
> 
> 1. `GitImport` provides information to import an application from a Git repository.
> 2. `ContainerImport` provides information to import a container image.
> 
> Other types will be ignored by the web console so that it is possible that operators install `type`s and downgrade the cluster when other source types are supported in the future.
> 
> #### `GitImport`
> 
> All previously supported samples opens a simplified "Import from Git" flow and prefill the Git repository.
> 
> When a `GitImport` sample is selected, the same form will be opened and prefilled.
> 
> 1. The Git `url` is required.
> 2. A Git `ref` is optional and can be used to clone a specific branch, tag or commit.
> 
> There is no other attribute supported at the moment. It supports only public Git repositories.
> 
> ```yaml
>   source:
>     type: GitImport
>     gitimport:
>       url: https://github.com/jboss-openshift/openshift-quickstarts
>       ref: master # optional
> ```
> 
> #### `ContainerImport`
> 
> When the user selects a `ContainerImport` sample, the **"Add"** > **"Container images"** page will be opened
> and the given container image is prefilled as "Image name from external registry".
> 
> Other options like the Resource type (Deployment, DeploymentConfig or Serverless Deployment) will
> be selected based on the cluster defaults and the user preferences.
> 
> ```yaml
>   source:
>     type: ContainerImport
>     containerimport:
>       image: registry.access.redhat.com/ubi8/ubi-minimal:8.8-860
> ```
>